src/iop/filmicrgb.c

```diff
@@ -1525,110 +1265,132 @@ void gui_init(dt_iop_module_t *self)
-  dt_bauhaus_widget_set_label(g->output_power, NULL, _("target power transfer function"));
-  gtk_box_pack_start(GTK_BOX(page3), g->output_power, FALSE, FALSE, 0);
+  g->output_power = dt_bauhaus_slider_new_from_params_box(self, "output_power");
   gtk_widget_set_tooltip_text(g->output_power, _("power or gamma of the transfer function\nof the display or color space.\n"
                                                  "you should never touch that unless you know what you are doing."));
-  g_signal_connect(G_OBJECT(g->output_power), "value-changed", G_CALLBACK(output_power_callback), self);
+
+  self->widget = gtk_box_new(GTK_ORIENTATION_VERTICAL, DT_BAUHAUS_SPACE);
+  dt_gui_add_help_link(self->widget, dt_get_help_url(self->op));
+
+  gtk_box_pack_start(GTK_BOX(self->widget), GTK_WIDGET(g->area), TRUE, TRUE, 0);
+  gtk_box_pack_start(GTK_BOX(self->widget), GTK_WIDGET(g->notebook), FALSE, FALSE, 0);
+  gtk_widget_show_all(GTK_WIDGET(gtk_notebook_get_nth_page(g->notebook, 0)));
+  dtgtk_justify_notebook_tabs(g->notebook);
 
   dt_iop_init_picker(&g->color_picker,
               self,
               DT_COLOR_PICKER_AREA,
               _iop_color_picker_get_set,
               _iop_color_picker_apply,
               _iop_color_picker_update);
-
-  dt_gui_add_help_link(self->widget, dt_get_help_url(self->op));
 }
 
+void gui_changed(dt_iop_module_t *self, GtkWidget *w, void *previous)
+{
+  dt_iop_filmicrgb_params_t *p = (dt_iop_filmicrgb_params_t *)self->params;
+  dt_iop_filmicrgb_gui_data_t *g = (dt_iop_filmicrgb_gui_data_t *)self->gui_data;
+
+  if (w == g->security_factor || w == g->grey_point_source || 
+      w == g->black_point_source || w == g->white_point_source)
+  {
+    const int reset = darktable.gui->reset;
+    darktable.gui->reset = 1;
+
+    if (w == g->security_factor || w == g->grey_point_source)
+    {
+      float prev = *(float*)previous;
+      if (w == g->security_factor)
+      {
+        float ratio = (p->security_factor - prev) / (prev + 100.0f);
+
+        float EVmin = p->black_point_source;
+        EVmin = EVmin + ratio * EVmin;
+
+        float EVmax = p->white_point_source;
+        EVmax = EVmax + ratio * EVmax;
+
+        p->white_point_source = EVmax;
+        p->black_point_source = EVmin;
+      }
+      else
+      {
+        float grey_var = log2f(prev / p->grey_point_source);
+        p->black_point_source = p->black_point_source - grey_var;
+        p->white_point_source = p->white_point_source + grey_var;
+      }
+
+      dt_bauhaus_slider_set_soft(g->white_point_source, p->white_point_source);
+      dt_bauhaus_slider_set_soft(g->black_point_source, p->black_point_source);
+    }
+
+    p->output_power =  logf(p->grey_point_target / 100.0f) / logf(-p->black_point_source / (p->white_point_source - p->black_point_source));
+
+    dt_bauhaus_slider_set_soft(g->output_power, p->output_power);
+  
+    darktable.gui->reset = reset;
+  }
+
+  gtk_widget_queue_draw(self->widget); // Is this needed?
```

Comment:
this is needed only if the drawing area needs a redraw, aka if the curve params have changed (white, grey, black, latitude, contrast, etc.)